frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts

```diff
@@ -198,6 +198,15 @@ export const sessionRecordingDataLogic = kea<sessionRecordingDataLogicType>([
                 },
             },
         ],
+        sessionPlayerDataLoading: [
+            false,
+            {
+                loadRecordingMetaSuccess: (_, { sessionPlayerData }) => {
+                    // Keep `sessionPlayerDataLoading` for when meta is loaded before snapshots finish loading
+                    return Object.keys(sessionPlayerData?.snapshotsByWindowId ?? {}).length === 0
+                },
+            },
+        ],
```

Comment:
Not sure how keen on this I am. We're overriding the built in loader reducer which feels error prone. Plus we could have a reverse race condition where it happens the other way.

I think we either need to break out the two loaders (one for metadata, one for snapshots) or have a unique reducer that records both loading states together so we can tell if we are "loadingMetadataOrSnapshots"

Reply:
True, this solution does rely on snapshots always taking a longer time. I'll go with the breaking up into two loaders method